### ISSUE TITLE
array-type vs prefer-array-literal

### ISSUE THREAD

**Issue by dashmug** [palantir/tslint#2946]
### Bug Report

- __TSLint version__: 5.4.3
- __TypeScript version__: 2.3.4
- __Running TSLint via__: WebStorm

#### TypeScript code being linted

I want `values` to be a mixed array of either an `object` or a `number`.

```ts
// Violates 'array-type' rule
const values: (object | number)[] = []
// Violates 'prefer-array-literal' rule
const values: Array<object | number> = []
```

with `tslint.json` configuration:

```json
{
  "extends": [
    "tslint:latest",
    "tslint-eslint-rules",
    "tslint-config-airbnb"
  ]
}
```

How can I write my code so I won't violate both rules? Are those two conflicting with each other and  I have to give up one rule?

**Comment by adidahiya** [comment]
It seems like `prefer-array-literal` just isn't compatible with the default config for the `array-type` rule. That rule isn't maintained by tslint, it's part of [tslint-microsoft-contrib](https://github.com/Microsoft/tslint-microsoft-contrib). You could set the `"array"` on `array-type` to get them to work together. You might want to file an issue on tslint-config-airbnb about this.

**Comment by iamakulov** [comment]
For everyone getting here from Google search, this can also be fixed without changing the `array-type` rule and reformatting array usages. Enable the `allow-type-parameters` option in `prefer-array-literal`:

```json
{
  "rules": {
    "prefer-array-literal": [true, { "allow-type-parameters": true }]
  }
}
```

**Comment by JoshuaKGoldberg** [comment]
Closing as external. Feel free to post an issue on [`tslint-microsoft-contrib`](https://github.com/microsoft/tslint-microsoft-contrib)!
